client/src/pages/UserAdminRegistration.tsx

```diff
@@ -0,0 +1,129 @@
+                    {success && <Alert variant="success">{success}</Alert>}
+                    <Form onSubmit={handleSubmit}>
+                        <Form.Group as={Row} controlId="formUsername">
+                            <Form.Label column sm={3}>Username</Form.Label>
+                            <Col sm={9}>
+                                <Form.Control
+                                    type="text"
+                                    name="username"
+                                    placeholder="Enter admin username"
+                                    value={formData.username}
+                                    onChange={handleChange}
+                                    required
+                                />
+                            </Col>
+                        </Form.Group>
+
+                        <Form.Group as={Row} controlId="formEmail">
+                            <Form.Label column sm={3}>Email</Form.Label>
+                            <Col sm={9}>
+                                <Form.Control
+                                    type="email"
+                                    name="email"
+                                    placeholder="user@example.org"
+                                    value={formData.email}
+                                    onChange={handleChange}
+                                    required
+                                />
+                            </Col>
+                        </Form.Group>
+
+                        <Form.Group as={Row} controlId="formPassword">
+                            <Form.Label column sm={3}>Password</Form.Label>
+                            <Col sm={9}>
+                                <Form.Control
+                                    type="password"
+                                    name="password"
+                                    placeholder="Enter admin password"
+                                    value={formData.password}
+                                    onChange={handleChange}
+                                    required
+                                />
+                            </Col>
+                        </Form.Group>
+
+                        <Form.Group as={Row} controlId="formConfirmPassword">
+                            <Form.Label column sm={3}>Confirm Password</Form.Label>
+                            <Col sm={9}>
+                                <Form.Control
+                                    type="password"
+                                    name="confirmPassword"
+                                    placeholder="Confirm admin password"
+                                    value={formData.confirmPassword}
+                                    onChange={handleChange}
+                                    required
+                                />
+                            </Col>
+                        </Form.Group>
```

Comment:
I think password confirmation isn't necessary when in this case, because we can assume that the super-admin is extra careful when creating admin accounts.
There's a whole debate over whether password confirmation is a good idea in the first place, both security-wise and also from a UI point of view.

If you want to keep it, that's fine. I guess it doesn't matter much.

Reply:
I considered it from a UI perspective. While the super-admin is an advanced user and may not necessarily need this feature, he is still human and prone to making typos, just like any regular user. For now, I’ve decided to keep it.